src/pages/ProfilePage/ProfilePage.jsx

```diff
@@ -61,41 +65,47 @@ const ProfilePage = () => {
   );
 
   const bodyContent = address ? (
-    <>
-      {!nftsClaimedByAUser.length && getEmptyProfilePageContent()}
-      {nftsClaimedByAUser.length !== 0 && (
-        <div>
-          <ProfileHeading>Claimed Missions</ProfileHeading>
-          <Stack
-            direction={{ xs: "column", sm: "row" }}
-            justifyContent="center"
-            alignItems="center"
-            // spacing={4}
-            flexWrap="wrap"
-            sx={{ my: 4 }}
-          >
-            {nftsClaimedByAUser.map((data, i) => (
-              <RoundedRect key={i}>
-                <img
-                  src={data.image_url}
-                  alt=""
-                  style={{ width: "200px", borderRadius: "5px" }}
-                />
-              </RoundedRect>
-            ))}
-          </Stack>
-        </div>
-      )}
-    </>
+    profileDataLoading ? (
+      <LoadingContainer>
+        <img src={loading} alt="Loading" style={{ width: '296px' }} />
+      </LoadingContainer>
+    ) : (
+      <>
+        {!nftsClaimedByAUser.length && getEmptyProfilePageContent()}
+        {nftsClaimedByAUser.length !== 0 && (
+          <div>
+            <ProfileHeading>Claimed Missions</ProfileHeading>
+            <Stack
+              direction={{ xs: 'column', sm: 'row' }}
+              justifyContent="center"
+              alignItems="center"
+              // spacing={4}
+              flexWrap="wrap"
+              sx={{ my: 4 }}
+            >
+              {nftsClaimedByAUser.map((data, i) => (
+                <RoundedRect key={i}>
+                  <img
```

Comment:
maybe create a separate image component and pass props since we are using them multiple places? create separate task if you think this will be feasible considering we'll be expanding it in future

Reply:
Done